src/workspace/src/logger_text/src/logger_text.cpp

```diff
@@ -125,38 +148,54 @@ void Writer::imuCallback(const sensor_msgs::Imu& imu){
 }
 
 void Writer::speedCallback(const nav_msgs::Odometry& speed){
-	double speed_threshold_kmh = 0.0;
-	if(ros::param::get("/logger_text/speed_threshold", speed_threshold_kmh)){
-		if(speed_threshold_kmh < 0 && speed_threshold_kmh > 100){
-			speed_threshold_kmh = 5.0;
-			ROS_ERROR("invalid speed threshold, defaulting to 5 Kmh");
-		}
-	}
-	else{
-		speed_threshold_kmh = 5.0;
-		ROS_INFO("no speed threshold parameter set, defaulting to 5 Kmh");
-	}	
-	
+
+	bool automaticMode = true; //temporary
 	
-	double current_speed = speed.twist.twist.linear.y;
-	if (kmh_Speed_list.size() < 120){
-		kmh_Speed_list.push_back(current_speed);
-	}
-	else{
-		kmh_Speed_list.pop_front();
-		kmh_Speed_list.push_back(current_speed);
-		average_speed = std::accumulate(kmh_Speed_list.begin(), kmh_Speed_list.end(), average_speed) / 120;
-		logger_service::GetLoggingStatus::Request request;
-		logger_service::GetLoggingStatus::Response response;
+	if(automaticMode){
+		speedThresholdKmh = getSpeedThreshold();
+
+		if(speedThresholdKmh < 0 && speedThresholdKmh > 100){
+			speedThresholdKmh = defaultSpeedThreshold;
+			std::string speed = std::to_string(speedThresholdKmh);
+			ROS_ERROR_STREAM("invalid speed threshold, defaulting to "<<speed<<" Kmh");
+		}
+		else if(speedThresholdKmh == 0.0){
+			speedThresholdKmh = defaultSpeedThreshold;
+			std::string speed = std::to_string(speedThresholdKmh);
```

Comment:
No need to convert to std::String to use ROS_ERROR_STREAM

PS: Redundant logic

Reply:
removed redundant logic